### PR TITLE
chore: pg_upgrade md5-to-scram migration

### PR DESCRIPTION
**Pull request by pcnc**
Please go the the `Preview` tab and select the appropriate sub-template:

* [Default](?expand=1&template=default.md)
* [Extension Upgrade](?expand=1&template=extension_upgrade.md)

### CODE REVIEW

File: ansible/files/admin_api_scripts/pg_upgrade_scripts/initiate.sh [hunk 1]
```diff
@@ -133,13 +133,26 @@ EOF
     for EXTENSION in "${EXTENSIONS_TO_DISABLE[@]}"; do
         EXTENSION_ENABLED=$(run_sql -A -t -c "SELECT EXISTS(SELECT 1 FROM pg_extension WHERE extname = '${EXTENSION}');")
         if [ "$EXTENSION_ENABLED" = "t" ]; then
+            echo "Exporting dependent objects for extension ${EXTENSION}, if they exist"
+            EXPORT_QUERY=$(cat << EOF
+    SELECT pg_get_functiondef(objid) || ';' AS function_definition
```
Comment (darora) [inline ansible/files/admin_api_scripts/pg_upgrade_scripts/initiate.sh]: Ideally we would have tests for this query

**Review by darora — APPROVED** [review]

File: ansible/files/admin_api_scripts/pg_upgrade_scripts/initiate.sh (hunk 1 above)
Comment (darora) [reply ansible/files/admin_api_scripts/pg_upgrade_scripts/initiate.sh]: Hmm does this capture multiple levels of dependencies?

e.g. view -> table -> function -> generated column uses plv8

Comment (darora) [reply ansible/files/admin_api_scripts/pg_upgrade_scripts/initiate.sh]: I'm guessing dropping the ext would cause the entire chain to be dropped?

Comment (pcnc) [reply ansible/files/admin_api_scripts/pg_upgrade_scripts/initiate.sh]: Yep, pretty much. Currently looking into a more involved way to detect these

**Pull request merged**